Fixes #35385 - Prevent auto-reset of password while editing

ofedoren opened the pull request.

ofedoren committed:
Fixes #35385 - Prevent auto-reset of password while editing

MariaAga reviewed (changes requested):
Looks good, left one comment. There is a bug that if a user clicks on "change the password" icon, closes the modal (without saving) and opens it again (with the same or different webhook) the password is editable. 
I would set the value to false in the `onClose` function for the modal or something

ofedoren commented:
Thanks, @MariaAga, updated. Should work now, but due to some limitations and just to not make a PR to the core, I needed to force users to use the buttons only to close the modal. Thus I've removed `x` mark in the header.

MariaAga commented on webpack/ForemanWebhooks/Routes/Webhooks/Components/WebhookForm/Components/ForemanFormikField.js:
```diff
@@ -56,10 +67,31 @@ const ForemanFormikField = ({
+              <span className="input-group-btn">
+                <a
+                  className="btn btn-default"
+                  href="#"
```
@ofedoren any reasons for keeping it an `a` tag and not changing to button?

ofedoren replied:
@MariaAga, my bad, forgot to change.

MariaAga commented:
Yup, works well now!

ofedoren commented:
Updated with button instead of link.

MariaAga reviewed (approved):
lgtm!

ofedoren commented:
Thanks, @MariaAga! Merging then :)
